internal/grpctest/tlogger.go

```diff
@@ -27,15 +27,16 @@ import (
 	"runtime"
 	"strconv"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
 	"google.golang.org/grpc/grpclog"
 )
 
-// TLogger serves as the grpclog logger and is the interface through which
+// tLoggerAtomic serves as the grpclog logger and is the interface through which
 // expected errors are declared in tests.
-var TLogger *tLogger
+var tLoggerAtomic atomic.Value
```

Comment:
@easwars, the logger is mutated only once, in the `init()` function of this package. The race usually happens in the `Update` method (e.g: [b/416700146#comment4](b/416700146#comment4)) when it calls grpclog.SetLoggerV2:
https://github.com/grpc/grpc-go/blob/6dfe07c8c3e44c50dd7821626de1d73ed44bc455/grpclog/loggerv2.go#L35-L41
Ideally `Update` should be called serially at the beginning of a test function. I've seen race conditions reported when a test doesn't wait for its goroutines to return before the test function exits. Using an atomic for `LoggerV2Impl` and i`DepthLoggerV2Impl` might hide those errors and make it appear as though the next test has logged something. Do you think we should really change this to use an atomic?

Reply:
Talked to @easwars and he mentioned that we can revert the changes to make the logger an atomic.